src/formats/farbfeld.zig

```diff
@@ -0,0 +1,109 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const mem = std.mem;
+const Allocator = std.mem.Allocator;
+const assert = std.debug.assert;
+const ImageUnmanaged = @import("../ImageUnmanaged.zig");
+const FormatInterface = @import("../FormatInterface.zig");
+const color = @import("../color.zig");
+const utils = @import("../utils.zig");
+const buffered_stream_source = @import("../buffered_stream_source.zig");
+
+pub const ReadError = ImageUnmanaged.ReadError;
+pub const WriteError = ImageUnmanaged.WriteError || error{BigDimensions};
+
+pub const Header = extern struct {
+    const width_size = 4;
+    const height_size = 4;
+    const size = 16;
+    const magic_value: [8]u8 = [8]u8{ 'f', 'a', 'r', 'b', 'f', 'e', 'l', 'd' };
+
+    width: u32 align(1),
+    height: u32 align(1),
+
+    fn encode(header: Header) [size]u8 {
+        var result: [size]u8 = undefined;
+        @memcpy(result[0..8], &magic_value);
+        std.mem.writeInt(u32, result[8..12], header.width, .big);
+        std.mem.writeInt(u32, result[12..16], header.height, .big);
+        return result;
+    }
+    comptime {
+        assert(size == 16);
```

Comment:
Should you test that `Header` is actually 16 bytes instead of testing to a constant?